compiler/main/arg.cpp

```diff
@@ -182,6 +182,12 @@ void usage(const ArgumentState* state,
             break;
 
           case 'P':
+            if (desc[i].location != 0)
+              printf("'%s'", ((std::string*) desc[i].location)->c_str());
+            else
+              printf("''");
+            break;
+
```

Comment:
definitely outside the scope of this PR, but it would be nice to change 'S' to use std::string too. Maybe open an issue for refactoring the arg parsing code to use more modern C++?

Reply:
Is there something you would want to see change about how it functions, besides the internals? I feel "if it ain't broke" about refactoring it, personally.

Reply:
Nothing specific in terms of the API, I mostly am just tired of dealing with very old C-style C++ code, which is true for much of the compiler

>  I feel "if it ain't broke" about refactoring it, personally.

I kinda tend to agree, but I think there is an aspect of "leave things better than you found it" too. Definitely not advocating you do anything in this regard for this PR.